src/modules/ModuleTradeShared/composable.pair-input.ts

```diff
@@ -24,6 +26,53 @@ export function useLocalStorageAddrsOrigin(key: string, isActive?: Ref<boolean>)
   return offable
 }
 
+export function useRouteAddrsOrigin({
+  baseToken,
+  additionalBaseToken,
+  isActive,
+}: {
+  baseToken?: Address
+  additionalBaseToken?: Address
+  isActive?: Ref<boolean>
+}) {
+  const params = {
+    tokenA: useRouteParams('tokenA'),
+    tokenB: useRouteParams('tokenB'),
+  }
+
+  return computed<TokensPair<Address | null>>({
+    get: () => {
+      if (isActive?.value) {
+        const addressOrNull = (x: unknown) => (typeof x === 'string' && isAddress(x) ? x : null)
+        const { tokenA, tokenB } = buildPair((type) => addressOrNull(params[type].value))
+
+        if (tokenA) {
+          if (tokenB) {
+            return { tokenA, tokenB }
+          }
+          if (baseToken && !areAddressesEqual(tokenA, baseToken)) {
+            return { tokenA, tokenB: baseToken }
+          }
+          if (additionalBaseToken) {
+            return { tokenA, tokenB: additionalBaseToken }
+          }
+          return { tokenA, tokenB: null }
+        }
+      }
+
+      return emptyPair()
+    },
+    set: (v) => {
+      if (v) {
+        params.tokenA.value = v.tokenA
+        params.tokenB.value = v.tokenB
+      } else {
+        params.tokenA.value = params.tokenB.value = null
+      }
```

Comment:
If the whole `computed` is `TokensPair<null | Address>`, then `v` is never nullable.

Reply:
Removed unnecessary condition